airgun/entities/new_contentview.py

```diff
@@ -0,0 +1,225 @@
+        view = self.navigate_to(self, 'Edit', entity_name=entity_name)
+        view.versions.search(version_name)
+        view.versions.table.row(version=version_name)['Status'].widget.wait_for_result()
+        view.flash.assert_no_error()
+        view.flash.dismiss()
+        return view.versions.table.row(version=version_name).read()
+
+    def publish_and_promote(self, entity_name, lce_name, values=None):
+        view = self.navigate_to(self, 'Publish', entity_name=entity_name)
+        if values:
+            view.fill(values)
+        # some type of breaks/pause needed in between to recognized current location before
+        # proceeding to next screen
+        view.lce(lce_name).fill(True)
+        view.next.click()
+        view.next.click()
+        view.progressbar.wait_for_result()
+        view = self.navigate_to(self, 'Edit', entity_name=entity_name)
+        # TODO: Not able to recognize table very well
+        view.versions.table[0]['Status'].widget.wait_for_result()
+        view.flash.assert_no_error()
+        view.flash.dismiss()
+        return view.versions.table[0].read()
+
+
+    def update(self, entity_name, values):
+        """Update existing content view"""
+        view = self.navigate_to(self, 'Edit', entity_name=entity_name)
+        # need a wait to recognize the loading is complete
+        # sleep works for now
+        time.sleep(3)
+        filled_values = view.fill(values)
+        view.flash.assert_no_error()
+        view.flash.dismiss()
+        return filled_values
+
+@navigator.register(NewContentViewEntity, 'All')
+class ShowAllContentViews(NavigateStep):
+    """Navigate to All Content Views screen."""
+
+    VIEW = NewContentViewTableView
+
+    def step(self, *args, **kwargs):
+        self.view.menu.select('Content', 'Content Views')
+
+
+@navigator.register(NewContentViewEntity, 'Create')
+class CreateContentView(NavigateStep):
+    """Navigate to Create content view."""
+
+    VIEW = NewContentViewCreateView
+
+    prerequisite = NavigateToSibling('All')
+
+    def step(self, *args, **kwargs):
+        self.parent.create_content_view.click()
+
+
+@navigator.register(NewContentViewEntity, 'Edit')
+class EditContentView(NavigateStep):
+    """Navigate to Edit Content View screen.
+
+    Args:
+        entity_name: name of content view
+    """
+
+    VIEW = NewContentViewEditView
+
+    def prerequisite(self, *args, **kwargs):
+        return self.navigate_to(self.obj, 'All')
+
+    def step(self, *args, **kwargs):
+        entity_name = kwargs.get('entity_name')
+        self.parent.search(entity_name)
+        self.parent.table.row(name=entity_name)['Name'].widget.click()
+
+@navigator.register(NewContentViewEntity, 'Delete')
+class DeleteContentView(NavigateStep):
+    """Navigate to Delete Content View screen by selecting appropriate action
+    on Edit Content View screen.
+
+    Args:
+        entity_name: name of content view
+    """
+
+    VIEW = NewContentViewRemoveView
```

Comment:
This is missing view ? Cant locate in the views.

Reply:
The renaming should fix this now.